web/models/stable_diffusion/utils.py

```diff
@@ -168,23 +167,27 @@ def map_device_to_name_path(device, key_combination=3):
         return device_mapping
 
     if "vulkan" in args.device:
-        name, args.device = map_device_to_name_path(args.device)
-        triple = get_vulkan_target_triple(name)
-        print(f"Found device {name}. Using target triple {triple}")
-        # set triple flag to avoid multiple calls to get_vulkan_triple_flag
-        if args.iree_vulkan_target_triple == "" and triple is not None:
-            args.iree_vulkan_target_triple = triple
+        device_name, args.device = map_device_to_name_path(args.device)
 
-        # use tuned models only in the case of rdna3 cards.
+        # set triple flag to avoid multiple calls to get_vulkan_triple_flag
         if not args.iree_vulkan_target_triple:
-            if triple is not None and "rdna3" not in triple:
-                args.use_tuned = False
-        elif "rdna3" not in args.iree_vulkan_target_triple:
+            triple = get_vulkan_target_triple(device_name)
+            if triple is not None:
+                args.iree_vulkan_target_triple = triple
+        print(
+            f"Found device {device_name}. Using target triple {args.iree_vulkan_target_triple}."
+        )
+
+        # use tuned models only in the case of stablediffusion/fp16 and rdna3 cards.
+        if (
+            args.variant != "stablediffusion"
+            or args.precision != "fp16"
+            or "rdna3" not in args.iree_vulkan_target_triple
+        ):
             args.use_tuned = False
```

Comment:
nit: add a warning/log similar to the one added below if use_tuned was set to true.

Reply:
Added same message.